File: src/types/system.ts
export interface AppInfo {
  name: string;
  version: string;
  environment: string;
}

export interface HealthStatus {
  status: string;
  database: string;
}

export interface DatabaseStatus {
  connected: boolean;
  path: string;
  schemaVersion: number;
}

export interface AppSettings {
  startWithWindows: boolean;
  closeToTray: boolean;
  systemNotifications: boolean;
  usageThresholdAlerts: boolean;
  predictionAlerts: boolean;
  warningThreshold: number;
  highThreshold: number;
  criticalThreshold: number;
  predictionAlertMinutes: number;
}

export interface AppSettingsSnapshot extends AppSettings {
  autostartRegistered: boolean | null;
  autostartAvailable: boolean;
  message: string | null;
}

File: src-tauri/src/models/system.rs
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct AppInfo {
    pub name: String,
    pub version: String,
    pub environment: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct HealthStatus {
    pub status: String,
    pub database: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct DatabaseStatus {
    pub connected: bool,
    pub path: String,
    pub schema_version: i64,
}
